rskj-core/src/test/java/co/rsk/rpc/modules/eth/EthModuleTest.java

```diff
@@ -52,10 +54,48 @@ public void callSmokeTest() {
                 .thenReturn(blockResult);
         when(blockResult.getBlock()).thenReturn(block);
 
-        byte[] hreturn = TypeConverter.stringToByteArray("hello");
+        byte[] hReturn = TypeConverter.stringToByteArray("hello");
+        ProgramResult executorResult = mock(ProgramResult.class);
+        when(executorResult.getHReturn())
+                .thenReturn(hReturn);
+
+        ReversibleTransactionExecutor executor = mock(ReversibleTransactionExecutor.class);
+        when(executor.executeTransaction(eq(blockResult.getBlock()), any(), any(), any(), any(), any(), any(), any()))
+                .thenReturn(executorResult);
+
+        EthModule eth = new EthModule(
+                null,
+                anyByte(),
+                null,
+                null,
+                executor,
+                retriever,
+                null,
+                null,
+                null,
+                new BridgeSupportFactory(
+                        null, null, null));
+
+        String expectedResult = TypeConverter.toUnformattedJsonHex(hReturn);
+        String actualResult = eth.call(args, "latest");
+
+        assertEquals(expectedResult, actualResult);
+    }
+
+    @Test
+    public void callWithoutReturn() {
+        Web3.CallArguments args = new Web3.CallArguments();
+        BlockResult blockResult = mock(BlockResult.class);
+        Block block = mock(Block.class);
+        ExecutionBlockRetriever retriever = mock(ExecutionBlockRetriever.class);
+        when(retriever.getExecutionBlock_workaround("latest"))
+                .thenReturn(blockResult);
+        when(blockResult.getBlock()).thenReturn(block);
+
+        byte[] hReturn = new byte[0];
```

Comment:
Could we have a test using an address without code? To be sure EthModule reacts with the expected behavior, even if we change the internal code in the future. In this test, we are presuming the hReturn will be the empty byte array.

Reply:
done